### PR TITLE
Update the "Updating LLVM" documentation

### PR DESCRIPTION
**Pull request by alexcrichton**
Make sure existing sections are up-to-date and then also add some words
about the recent convention we developed for updating LLVM versions with
respect to branch naming as well.

### CODE REVIEW

**Comment by alexcrichton** [comment]
r? @cuviper

I've also updated [README.md in rust-lang/llvm-project](https://github.com/rust-lang/llvm-project) to point to this documentation about updating.

File: src/codegen/updating-llvm.md [hunk 1]
```diff
@@ -70,27 +75,18 @@ through each in detail.
...
-
-4. Prepare a commit to rust-lang/rust
-
-  * Update `src/llvm-project`
-  * Update `compiler-builtins` crate in `Cargo.lock` (if necessary)
-
-5. Build your commit. Make sure you've committed the previous changes to ensure
+3. Build the new LLVM in the `rust` repository. To do this you'll want to update
+   the `src/llvm-project` repository to your branch and the revision you've
+   created. Make sure you've committed changes to `src/llvm-project` to ensure
```
Comment (cuviper) [inline src/codegen/updating-llvm.md]: We should make sure to update the branch name in `.gitmodules` too, as this enables easy `git submodules update --remote src/llvm-project`.

File: src/codegen/updating-llvm.md [hunk 2]
```diff
@@ -43,7 +46,9 @@ to the branch we're already using. The steps for this are:
 4. Check out the appropriate branch (typically named `rustc/a.b-yyyy-mm-dd`)
 5. Cherry-pick the upstream commit onto the branch
 6. Push this branch to your fork
-7. Send a Pull Request to rust-lang/llvm-project to the same branch as before
+7. Send a Pull Request to rust-lang/llvm-project to the same branch as before.
+   Be sure to reference the Rust and/or LLVM issue that you're fixing in the PR
+   description.
 8. Wait for the PR to be merged
 9. Send a PR to rust-lang/rust updating the `src/llvm-project` submodule with
    your bugfix
```
Comment (cuviper) [inline src/codegen/updating-llvm.md]: This is another place to mention `git submodule update --remote src/llvm-project`, assuming the branch name in `.gitmodules` is current.

**Comment by alexcrichton** [comment]
Excellent points, updated!

**Pull request merged**